featurebyte/service/feature_list.py

```diff
@@ -82,9 +82,11 @@ async def validate_feature_list_version_and_namespace_consistency(
         )
 
     feature_namespace_ids = []
-    for feature_id in feature_list.feature_ids:
-        feature = await feature_service.get_document(document_id=feature_id)
-        feature_namespace_ids.append(feature.feature_namespace_id)
```

Comment:
Feature retrieval one by one is expensive as it
* makes multiple mongo queries
* query results will be deserialized to feature model.

Revise this part to optimize time taken to validate feature list.